src/OpenTelemetry/Internal/OpenTelemetrySdkEventSource.cs

```diff
@@ -195,7 +195,7 @@ public void ShutdownEvent(int spansLeftUnprocessed)
         [Event(3, Message = "Exporter returned error '{0}'.", Level = EventLevel.Warning)]
         public void ExporterErrorResult(ExportResult exportResult)
         {
-            this.WriteEvent(3, exportResult.ToString());
+            this.WriteEvent(3, exportResult);
```

Comment:
Does this fix the behavior we're seeing here https://github.com/open-telemetry/opentelemetry-dotnet/issues/3619#issuecomment-1234724122? If so, we should get this landed.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/598 and old PR doing similar stuff.

Reply:
@alanwest, I don't have a straightforward repo to confirm if this is the issue seen in #3619. 

@cijothomas, I verified that my change works, and can confirm that the `ExportResult` enum is successfully converted to string.
I don't think the `[NonEvent]` pattern in #598 is necessary. But I'm happy to change if that's the preferred pattern. :)

Reply:
nonevent pattern is to save string alloc if none listening.

Does this PR's change create the string if none is listening?

Reply:
changed to `NonEvent` pattern